docs/summer-of-code-ideas/2019-JiradetOunjai.md

```diff
@@ -0,0 +1,140 @@
+# Google Summer of Code 2019 Report - Jiradet Ounjai
+
+The Google Summer of Code program has turned out to be a great learning opportunity for me. I gain fresh knowledge from my fantastic mentors through the discussions and the intense code reviews we had. I would like to thank my mentors for the time and energy you put into helping me.
+
+I have made this document to summarize the work done for my Google Summer of Code project *Improve Shading Language Support in GraphicsFuzz*. 
+
+
+## Personal Links
+[Github](https://github.com/jiradeto) - [Linkedin](https://www.linkedin.com/in/jiradeto/)
+
+
+## Project Link
+[Improve Shading Language Support in GraphicsFuzz](https://summerofcode.withgoogle.com/projects/#4859963594244096)
+
+
+## Mentors
+Alastair Donaldson, Paul Thomson
+
+
+## Deliverables
+My project idea focused on providing a new shading language support in GraphicsFuzz. To accomplish the goal, I have enhanced the tool in various aspects which could be elaborated as follow:
+
+- [Add support for GLSL Built-In Functions](#add-support-for-glsl-built-in-functions)
+- [Add new ways to generate an opaque expression](#add-new-ways-to-generate-an-opaque-expression)
+- [New sample shaders](#new-sample-shaders)
+- [Enhancement for Reducer](#enhancement-for-reducer)
+- [Use GraphicsFuzz to find bugs in other tools](#use-graphicsfuzz-to-find-bugs-in-other-tools)
+- [New shader generator tool](#new-shader-generator-tool)
+
+
+
+### Add support for GLSL Built-In Functions 
+GLSL built-in functions are functions that available for use in a shader. GraphicsFuzz donates these built-ins into the unreachable statements so that they will not be actually executed. By providing a support for the GLSL built-ins, we teach GraphicsFuzz how to call functions appropriately by ensuring that the required function arguments are matched with the GLSL specifications. Moreover, we have to check that the built-in function being injected into a shader is compatible with the given shading language version. The following PRs have provided a support for the GLSL built-in functions.
+
+[#554](https://github.com/google/graphicsfuzz/pull/554): Add support for angle and trigonometric built-in functions
+
+[#555](https://github.com/google/graphicsfuzz/pull/555): Add support for built-in Matrix functions 
+
+[#581](https://github.com/google/graphicsfuzz/pull/581): Refactor geometric builtins 
+
+[#585](https://github.com/google/graphicsfuzz/pull/585): Refactor floating-point pack and unpack built-ins
+
+[#592](https://github.com/google/graphicsfuzz/pull/592): Complete common builtins
+
+
+### Add new ways to generate an opaque expression
```

Comment:
"Opaque expression, statement uninterpreted and thus non optimizable by compilers" - could be worked on more in terms of English.

Reply:
Fixed